diff --git a/conditional_fields/js/conditional_fields.js b/conditional_fields/js/conditional_fields.js
index fafbeb7..b7f1d53 100644
--- a/conditional_fields/js/conditional_fields.js
+++ b/conditional_fields/js/conditional_fields.js
@@ -118,23 +118,25 @@ Drupal.behaviors.conditionalFields = {
     if (typeof conditionalFields === 'undefined' || typeof conditionalFields.effects === 'undefined') {
       return;
     }
-    // Override state change handlers for dependents with special effects.
-    $.each($(document).data('events'), function(i, events) {
-      if (i.substring(0, 6) === 'state:') {
-        var originalHandler = events[0].handler;
-        events[0].handler = function(e) {
-          var effect = conditionalFields.effects['#' + e.target.id];
-          if (typeof effect !== 'undefined') {
-            var effectEvent = i + '-' + effect.effect;
-            if (typeof $(document).data('events')[effectEvent] !== 'undefined') {
-              $(e.target).trigger({ type : effectEvent, trigger : e.trigger, value : e.value, effect : effect.options });
-              return;
+    // Override state change handlers for dep;endents with special effects.
+    if ($(document).data('events') !== undefined) {
+      $.each($(document).data('events'), function (i, events) {
+        if (i.substring(0, 6) === 'state:') {
+          var originalHandler = events[0].handler;
+          events[0].handler = function (e) {
+            var effect = conditionalFields.effects['#' + e.target.id];
+            if (typeof effect !== 'undefined') {
+              var effectEvent = i + '-' + effect.effect;
+              if (typeof $(document).data('events')[effectEvent] !== 'undefined') {
+                $(e.target).trigger({type: effectEvent, trigger: e.trigger, value: e.value, effect: effect.options});
+                return;
+              }
             }
+            originalHandler(e);
           }
-          originalHandler(e);
         }
-      }
-    });
+      });
+    }
   }
 };
 
