diff --git a/docroot/sites/all/modules/contrib/pki_authentication/pki_authentication.module b/docroot/sites/all/modules/contrib/pki_authentication/pki_authentication.module
index 4e2363f8f..5a842f265 100644
--- a/docroot/sites/all/modules/contrib/pki_authentication/pki_authentication.module
+++ b/docroot/sites/all/modules/contrib/pki_authentication/pki_authentication.module
@@ -256,7 +256,6 @@ function pki_authentication_form_alter(&$form, &$form_state, $form_id) {
       // $lang = (empty($account->language)) ? 'und' : $account->language;
       $lang = LANGUAGE_NONE;
       if ($form['#user']->uid == $user->uid and empty($user->pki_authentication_pki_string[$lang][0]['value'])) {
-        // $url = variable_get('pki_authentication_base_root', '') . '/' . url(variable_get('pki_authentication_pki_extraction_path', 'misc/pki_authentication/') . 'extract_pki.php?form=' . $form_id . "&uid=" . $user->uid, array('external' => TRUE));
         global $base_url;
         $cac_base_root = variable_get('pki_authentication_base_root', $base_url);
         $url = variable_get('gn2_cac_url', $cac_base_root) . url(variable_get('pki_authentication_pki_extraction_path', 'misc/pki_authentication/') . 'extract_pki.php?form=' . $form_id . "&uid=" . $user->uid, array('external' => TRUE));
@@ -469,7 +468,7 @@ function pki_authentication_user_edit_submit($form, &$form_state) {
     $user_data = (empty($form_state['values']['user_data']))? array() : @unserialize($form_state['values']['user_data']);
     if (!empty($user_data['init'])) {
       $account->pki_authentication_pki_string[$lang][0]['value'] = (empty($user_data['init']))? '' : check_plain($user_data['init']);
-      $account = user_save((object) array('uid' => $account->uid), (array) $account);
+      $account = user_save($account);
     }
   }
 }
