diff --git a/autologout.module b/autologout.module
index 4db5720..ada983c 100644
--- a/autologout.module
+++ b/autologout.module
@@ -325,7 +325,7 @@ function _autologout_user_uid_timeout_validate($element, &$form_state) {
 function autologout_user_profile_submit(&$form, &$form_state) {
   global $user;
   $current_uid = $user->uid;
-  $userid = $form_state['user']->uid;
+  $userid = $form['#user']->uid;
   $access = FALSE;
 
   if ((user_access('change own logout threshold') && $current_uid == $userid) || user_access('administer autologout')) {
@@ -335,7 +335,7 @@ function autologout_user_profile_submit(&$form, &$form_state) {
   // Access is reused here as a security measure. Not only will the element not
   // display but wont sumbit without access.
   if ($access) {
-    $userid = $form_state['user'] -> uid;
+    $userid = $form['#user']->uid;
     $val = $form_state['values']['autologout_user_' . $userid];
     variable_set('autologout_user_' . $userid, $val);
   }
