diff --git a/userpoints/views/userpoints_views_handler_field_category.inc b/userpoints/views/userpoints_views_handler_field_category.inc
index de1901c..3624b4e 100644
--- a/views/userpoints_views_handler_field_category.inc
+++ b/views/userpoints_views_handler_field_category.inc
@@ -44,10 +44,10 @@ function document_self_tokens(&$tokens) {
   }
 
   function add_self_tokens(&$tokens, $item) {
-    $tokens['[' . $this->options['id'] . '-tid' . ']'] = $item['tid'];
-    $tokens['[' . $this->options['id'] . '-name' . ']'] = $item['name'];
-    $tokens['[' . $this->options['id'] . '-vid' . ']'] = $item['vid'];
-    $tokens['[' . $this->options['id'] . '-vocabulary' . ']'] = $item['vocabulary'];
+    $tokens['[' . $this->options['id'] . '-tid' . ']'] = isset($item['tid']) ? $item['tid'] : NULL;
+    $tokens['[' . $this->options['id'] . '-name' . ']'] = isset($item['name']) ? $item['name'] : NULL;
+    $tokens['[' . $this->options['id'] . '-vid' . ']'] = isset($item['vid']) ? $item['vid'] : NULL;
+    $tokens['[' . $this->options['id'] . '-vocabulary' . ']'] = isset($item['vocabulary']) ? $item['vocabulary'] : NULL;
   }
 }
 
