diff --git features/includes/features.menu.inc features/includes/features.menu.inc
index c883e6a..37087a3 100644
--- features/includes/features.menu.inc
+++ features/includes/features.menu.inc
@@ -92,7 +92,7 @@ function menu_custom_features_export_render($module, $data) {
       ->execute()
       ->fetchAssoc();
     if ($row) {
-      $export = features_var_export($row, '  ');
+      $export = features_var_export($row, "  /* $menu_name */ ");
       $code[] = "  // Exported menu: {$menu_name}.";
       $code[] = "  \$menus['{$menu_name}'] = {$export};";
       $translatables[] = $row['title'];
