diff --git a/password_policy.module b/password_policy.module
index 1258c29..e30dc11 100644
--- a/password_policy.module
+++ bpassword_policy.module
@@ -244,6 +244,14 @@ function password_policy_user_load($users) {
   foreach ($users as $uid => $user) {
     $user->force_password_change = _password_policy_is_password_change_forced($uid);
     if (empty($user->force_password_change)) {
+      $history = db_select('password_policy_history', 'pp', array('target' => 'slave'))
+        ->fields('pp', array(
+            'created',
+          ))
+        ->condition('uid', $user->uid)
+        ->execute()
+        ->fetchField();
+        
       $blocked = db_select('password_policy_expiration', 'p', array('target' => 'slave'))
         ->fields('p', array(
           'blocked',
@@ -251,7 +259,7 @@ function password_policy_user_load($users) {
         ->condition('uid', $user->uid)
         ->execute()
         ->fetchField();
-      if (!empty($blocked)) {
+      if (!empty($blocked) && $history < $blocked) {
         if ($blocked <= time()) {
           $user->force_password_change = 1;
         }
